cfme/tests/infrastructure/test_quota.py

```diff
@@ -122,3 +194,45 @@ def test_quota(appliance, provider, setup_provider, custom_prov_data, vm_name, a
     provision_request = appliance.collections.requests.instantiate(request_description)
     provision_request.wait_for_request(method='ui')
     assert provision_request.row.reason.text == "Quota Exceeded"
+
+
+@pytest.mark.parametrize(
+    ['set_parent_tenant_quota', 'custom_prov_data', 'extra_msg', 'approve'],
+    [
+        [('cpu', '2'), {'hardware': {'num_sockets': '8'}}, '', False],
+        [('storage', '0.01'), {}, '', False],
+        [('memory', '2'), {'hardware': {'memory': '4096'}}, '', False],
+        [('vm', '1'), {'catalog': {'num_vms': '4'}}, '###', True]
+    ],
+    indirect=['set_parent_tenant_quota'],
+    ids=['max_cpu', 'max_storage', 'max_memory', 'max_vms']
+)
+def test_user_quota_diff_groups(request, appliance, provider, setup_provider, new_user,
+                                set_parent_tenant_quota, extra_msg, custom_prov_data, approve,
+                                prov_data, vm_name, template_name):
+    """prerequisite: Provider should be added
+
+    steps:
+
+    1. Create three tenants
+    2. Create Three new groups
+    3. Three groups should be assigned to three different tenants
+    4. Create new user
+    5. User should be member of three groups
+    6. Assign quota for three tenants(like ('cpu', '2') for three tenants at a time)
+    7. Provision VM with more than assigned quota
+    """
+    with new_user:
+        recursive_update(prov_data, custom_prov_data)
```

Comment:
Please log prov_data after this recursive update, so its clear when debugging what options are given for the provisioning.